backend/typescript/rest/schedulingRoutes.ts

```diff
@@ -0,0 +1,144 @@
+import { Router } from "express";
+
+import { isAuthorizedByRole } from "../middlewares/auth";
+import {
+  createSchedulingDtoValidator,
+  updateSchedulingDtoValidator,
+} from "../middlewares/validators/schedulingValidators";
+import nodemailerConfig from "../nodemailer.config";
+import AuthService from "../services/implementations/authService";
+import EmailService from "../services/implementations/emailService";
+import UserService from "../services/implementations/userService";
+import SchedulingService from "../services/implementations/schedulingService";
+import IUserService from "../services/interfaces/userService";
+import IAuthService from "../services/interfaces/authService";
+import IEmailService from "../services/interfaces/emailService";
+import ISchedulingService from "../services/interfaces/schedulingService";
+import { SchedulingDTO } from "../types";
+import { sendResponseByMimeType } from "../utilities/responseUtil";
+
+const schedulingRouter: Router = Router();
+
+// Commenting out authorization for now
+// schedulingRouter.use(isAuthorizedByRole(new Set(["Admin"])));
+
+const userService: IUserService = new UserService();
+const emailService: IEmailService = new EmailService(nodemailerConfig);
+const authService: IAuthService = new AuthService(userService, emailService);
+const schedulingService: ISchedulingService = new SchedulingService();
+
+/* Get all schedulings, optionally filter by:
+  - id, through URI (ex. /scheduling/1)
+  - donorId, through query param (ex. /scheduling/?donorId=1)
+*/
+schedulingRouter.get("/:id?", async (req, res) => {
+  const id: number | null = Number(req.params.id);
+  const donorId: number | null = Number(req.query.donorId);
+
+  const contentType = req.headers["content-type"];
+
+  if (id && donorId) {
+    await sendResponseByMimeType(res, 400, contentType, [
+      {
+        error: "Cannot query by both id and donorId.",
+      },
+    ]);
+    return;
+  }
+
+  if (!id && !donorId) {
+    try {
+      const schedulings = await schedulingService.getSchedulings();
+      await sendResponseByMimeType<SchedulingDTO>(
+        res,
+        200,
+        contentType,
+        schedulings,
+      );
+    } catch (error) {
+      await sendResponseByMimeType(res, 500, contentType, [
+        {
+          error: error.message,
+        },
+      ]);
+    }
+    return;
+  }
+
+  if (id) {
+    if (typeof id !== "number") {
```

Comment:
Also do we need to make this check since id could be either `null or number` so if passes through the first if statement, doesnt that mean it would be number now?

Reply:
Good catch. It's been changed so that the purpose of the double check to is to provide more meaningful error messages!